contrib/platform/src/com/sun/jna/platform/win32/User32.java

```diff
@@ -2484,5 +2487,329 @@ HANDLE SetWinEventHook(int eventMin, int eventMax, HMODULE hmodWinEventProc, Win
+	 *         combined with the specified virtual key to form a single character.
+	 *         However, the buffer may contain more characters than the return value
+	 *         specifies. When this happens, any extra characters are invalid and
+	 *         should be ignored.</dd>
+	 *         </dl>
+	 * 
+	 */
+	int ToUnicodeEx(UINT wVirtKey, UINT wScanCode, byte[] lpKeyState, char[] pwszBuff, int cchBuff, UINT wFlags,
+			HKL dwhkl);
+
+	/**
+	 * Loads a string resource from the executable file associated with a specified
+	 * module, copies the string into a buffer, and appends a terminating null
+	 * character.
+	 * 
+	 * @param hInstance    A handle to an instance of the module whose executable
+	 *                     file contains the string resource. To get the handle to
+	 *                     the application itself, call the GetModuleHandle function
+	 *                     with NULL.
+	 * @param uID          The identifier of the string to be loaded.
+	 * @param lpBuffer     The buffer is to receive the string. Must be of
+	 *                     sufficient length to hold a pointer (8 bytes).
+	 * @param cchBufferMax The size of the buffer, in characters. The string is
+	 *                     truncated and null-terminated if it is longer than the
+	 *                     number of characters specified. If this parameter is 0,
+	 *                     then lpBuffer receives a read-only pointer to the
+	 *                     resource itself.
+	 * @return If the function succeeds, the return value is the number of
+	 *         characters copied into the buffer, not including the terminating null
+	 *         character, or zero if the string resource does not exist. To get
+	 *         extended error information, call GetLastError.
+	 * 
+	 */
+	int LoadString(HINSTANCE hInstance, UINT uID, LPSTR lpBuffer, int cchBufferMax);
+
+	/**
+	 * Loads a string resource from the executable file associated with a specified
+	 * module and return a read-only pointer to the resource itself.
+	 * 
+	 * @param hInstance    A handle to an instance of the module whose executable
+	 *                     file contains the string resource. To get the handle to
+	 *                     the application itself, call the GetModuleHandle function
+	 *                     with NULL.
+	 * @param uID          The identifier of the string to be loaded.
+	 * @param lpBuffer     The buffer is to receive the string. Must be of
+	 *                     sufficient length to hold a pointer (8 bytes).
+	 * @param cchBufferMax Always pass 0.
+	 * @return If the function succeeds, the return value is the number of
+	 *         characters available the pointer, not including the terminating null
+	 *         character, or zero if the string resource does not exist. To get
+	 *         extended error information, call GetLastError.
+	 * 
+	 */
+	int LoadString(HINSTANCE hInstance, UINT uID, PointerByReference lpBuffer, int cchBufferMax);
```

Comment:
Why a `PointerByReference`?

Reply:
This function has two distinct call semantics. This is the case where the pointer to the memory is placed into the PointerByReference. See Javadoc.